### PR TITLE
DEP: remove extradoc in _distn_infrastructure

### LINKED ISSUE #15818: DEP: remove extradoc keyword in _distn_infrastructure
Documented as deprecated since https://github.com/scipy/scipy/commit/eb90ace863f0538dcacb67bcf246918037dc298b (scipy 0.10.0!), now with a warning since  #15748 / #15799.

### PR DESCRIPTION
**Pull request by jyuv**
#### Reference issue

Closes #15818 

#### What does this implement/fix?

Remove extradoc deprecated keyword from rv_continuous & rv_discrete classes

### CODE REVIEW

**Comment by h-vetinari** [comment]
Thanks for all your PRs!

You're a bit early with this one - this is scheduled only for 1.11. 😅

**Comment by rgommers** [comment]
Same comment as on your other PR; I'll close now and this can be reopened once 1.11.0 development starts.

**Comment by j-bowhay** [comment]
Reopening now development has started for 1.11.0

**Review by mdhaber — APPROVED** [review]

**Comment by mdhaber** [comment]
@j-bowhay would you double-check me on this? All the removals look OK to me, and I'm not seeing any other occurrences of `extradoc` left.

File: scipy/stats/tests/test_distributions.py [hunk 1]
```diff
@@ -5985,7 +5985,7 @@ def test_540_567():
 
 def test_regression_ticket_1316():
```
Comment (j-bowhay) [inline scipy/stats/tests/test_distributions.py:5986]: Is this still needed?

**Review by j-bowhay — COMMENTED** [review]
The removal looks good, I have pulled the branch locally and grepped to confirm no remaining instances of extradoc. I just had one query about the regression test.

File: scipy/stats/tests/test_distributions.py (hunk 1 above)
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5986]:
```suggestion
```

File: scipy/stats/tests/test_distributions.py [hunk 2]
```diff
@@ -5985,7 +5985,7 @@ def test_540_567():
...
     # The following was raising an exception, because _construct_default_doc()
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5987]:
```suggestion
```

File: scipy/stats/tests/test_distributions.py [hunk 3]
```diff
@@ -5985,7 +5985,7 @@ def test_540_567():
...
-    # did not handle the default keyword extradoc=None.  See ticket #1316.
+    # See ticket #1316.
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5988]:
```suggestion
```

File: scipy/stats/tests/test_distributions.py [hunk 4]
```diff
@@ -5985,7 +5985,7 @@ def test_540_567():
...
     stats._continuous_distns.gamma_gen(name='gamma')
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5989]:
```suggestion
```

File: scipy/stats/tests/test_distributions.py [hunk 5]
```diff
@@ -5985,7 +5985,7 @@ def test_540_567():
...
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5990]:
```suggestion
```

File: scipy/stats/tests/test_distributions.py [hunk 6]
```diff
@@ -5985,7 +5985,7 @@ def test_540_567():
...
```
Comment (mdhaber) [inline scipy/stats/tests/test_distributions.py:5991]:
```suggestion
```

**Review by mdhaber — APPROVED** [review]
You're right, it looks like this is no longer needed.

File: scipy/stats/tests/test_distributions.py (hunk 1 above)
Comment (mdhaber) [reply scipy/stats/tests/test_distributions.py:5986]: Nope, I'll remove it and merge. Thanks!

**Comment by mdhaber** [comment]
Thanks @jyuv @j-bowhay!

**Pull request merged**